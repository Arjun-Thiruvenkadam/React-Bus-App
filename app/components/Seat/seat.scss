.order {
  font-family: 'Montserrat', sans-serif;
  font-size: medium;
  background-color: white;
  width: 108px;
  display: inline-block;
  padding: 0px 15px;
  border: black solid 2px;
  margin: 10px;
  position: relative;
  transition-property: transform;
  transition-duration: 0.3s;
}

.order .tooltipText {
  position: absolute;
  visibility: hidden;
  width: max-content;
  z-index: 1;
  background-color: whitesmoke;
  border: 2px solid black;
  opacity: 0;
  transition: opacity 0.6s;
  line-height: 4px;
  bottom: 110%;
}

.order:hover {
  background-color: wheat;
}

.order:hover .tooltipText {
  visibility: visible;
  opacity: 1;
}

.order .tooltipText i {
  position: absolute;
  top: 100%;
  margin-left: -12px;
  width: 24px;
  height: 12px;
  overflow: hidden;
}

.order .tooltipText i::after {
  content: '';
  position: absolute;
  width: 12px;
  height: 12px;
  transform: translate(-50%, -50%) rotate(45deg);
  background-color: #eeeeee;
  box-shadow: 0 1px 8px rgba(0, 0, 0, 0.5);
}

.red {
  background-color: #ff5252;
}

.red:hover {
  background-color: #ff8a80;
}

.booking {
  background-color: #00e676;
  transform: scale(1.1, 1.1);
}

.booking:hover {
  background-color: #69f0ae;
}

.hidden {
  display: none;
}
