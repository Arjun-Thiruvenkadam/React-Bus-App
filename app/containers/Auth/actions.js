/*
 *
 * Auth actions
 *
 */

import * as actionTypes from './constants';

export function inputChange(event, controlName) {
  return {
    type: actionTypes.INPUT_CHANGE,
    event,
    controlName,
  };
}

export function submit(event) {
  return {
    type: actionTypes.SUBMIT,
    event,
  };
}

export function startLoading() {
  return {
    type: actionTypes.START_LOADING,
  };
}

export function stopLoading() {
  return {
    type: actionTypes.STOP_LOADING,
  };
}

export function authSuccess(name, token) {
  return {
    type: actionTypes.AUTH_SUCCESS,
    name,
    token,
  };
}

export function switchAuthMode(){
  return{
    type: actionTypes.SWITCH_AUTHMODE,
  }
}

export function logout(){
  return{
    type: actionTypes.LOGOUT,
  }
}

export function invalid(){
  return{
    type: actionTypes.INVALID,
  }
}
