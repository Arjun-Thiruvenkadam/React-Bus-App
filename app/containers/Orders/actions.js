/*
 *
 * Orders actions
 *
 */

import * as actionTypes from './constants';

export function bookTicket() {
  return {
    type: actionTypes.BOOK_TICKET,
  };
}

export function showSpinner() {
  return {
    type: actionTypes.SHOW_SPINNER,
  };
}

export function hideSpinner() {
  return {
    type: actionTypes.HIDE_SPINNER,
  };
}

export function saveTickets(tickets) {
  return {
    type: actionTypes.SAVE_TICKETS,
    tickets,
  };
}

export function cancelOrder() {
  return {
    type: actionTypes.CANCEL_ORDER,
  };
}

export function selectTicket(ticketId) {
  return {
    type: actionTypes.SELECT_TICKET,
    ticketId,
  };
}

export function fetchData() {
  return {
    type: actionTypes.FETCH_DATA,
  };
}

export function completeBooking() {
  return {
    type: actionTypes.COMPLETE_BOOKING,
  };
}

export function clearSelection() {
  return {
    type: actionTypes.CLEAR_SELECTION,
  };
}

export function update() {
  return {
    type: actionTypes.UPDATE,
  };
}
