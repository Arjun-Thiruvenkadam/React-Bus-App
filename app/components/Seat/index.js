/**
 *
 * Seat
 *
 */

import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import './seat.scss';

const Seat = props => {
  let styleClasses = ['order'];
  let tooltipClass = 'hidden';
  let onClickHandler = props.clicked;

  if (props.status === 'closed') {
    tooltipClass = 'tooltipText';
    styleClasses.push('red');
    onClickHandler = null;
  } else if (props.status === 'booking') {
    styleClasses.push('booking');
  }

  return (
    <Fragment>
      <div className={styleClasses.join(' ')} onClick={onClickHandler}>
        <p>SEAT : {props.id}</p>
      </div>
    </Fragment>
  );
};

Seat.propTypes = {
  name: PropTypes.string,
  status: PropTypes.string,
  id: PropTypes.number,
  clicked: PropTypes.func,
};

export default Seat;
