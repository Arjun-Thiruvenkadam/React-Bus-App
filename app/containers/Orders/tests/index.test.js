import React from 'react';
import { render } from 'react-testing-library';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import { browserHistory } from 'react-router-dom';
import { MemoryRouter } from 'react-router';
import { create } from 'react-test-renderer';
import ConnectedOrders, { Orders, mapDispatchToProps } from '../index';
import configureStore from '../../../configureStore';
import * as actions from '../actions';
import Seat from '../../../components/Seat/index';

describe('<Orders />', () => {
  let store;

  beforeAll(() => {
    store = configureStore({}, browserHistory);
  });

  it('should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <Provider store={store}>
        <MemoryRouter>
          <IntlProvider locale="en">
            <ConnectedOrders />
          </IntlProvider>
        </MemoryRouter>
      </Provider>,
    );
    expect(firstChild).toMatchSnapshot();
  });

  it('should render and match the snapshot with spinner', () => {
    const {
      container: { firstChild },
    } = render(
      <Provider store={store}>
        <MemoryRouter>
          <IntlProvider locale="en">
            <Orders
              showSpinner
              tickets={[{ ticketId: 1, status: 'open' }]}
              selectedTickets={[{ ticketId: 1, status: 'open' }]}
              ordering={false}
              shouldUpdate={false}
              authenticated
              fetchDataHandler={() => {}}
            />
          </IntlProvider>
        </MemoryRouter>
      </Provider>,
    );
    expect(firstChild).toMatchSnapshot();
  });

  it('<Seat> props.onClick is called on click ', () => {
    const fn = jest.fn();
    const tree = create(
      <IntlProvider locale="en">
        <Orders
          showSpinner={false}
          tickets={[{ ticketId: 1, status: 'open' }]}
          selectedTickets={[{ ticketId: 1, status: 'open' }]}
          ordering={false}
          shouldUpdate={false}
          authenticated
          fetchDataHandler={() => {}}
          selectTicketHandler={fn}
        />
      </IntlProvider>,
    );
    const SeatComponent = tree.root.findByType(Seat);
    SeatComponent.props.clicked();
    expect(fn.mock.calls.length).toBe(1);
  });

  describe('mapDispatchToProps', () => {
    describe('bookHandler', () => {
      it('should be injected', () => {
        const dispatch = jest.fn();
        const result = mapDispatchToProps(dispatch);
        expect(result.bookHandler).toBeDefined();
      });

      it('should dispatch bookTicket when called', () => {
        const dispatch = jest.fn();
        const result = mapDispatchToProps(dispatch);
        result.bookHandler();
        expect(dispatch).toHaveBeenCalledWith(actions.bookTicket());
      });
    });

    describe('selectTicketHandler', () => {
      it('should be injected', () => {
        const dispatch = jest.fn();
        const result = mapDispatchToProps(dispatch);
        expect(result.selectTicketHandler).toBeDefined();
      });

      it('should dispatch selectTicket when called', () => {
        const dispatch = jest.fn();
        const result = mapDispatchToProps(dispatch);
        const data = 1;
        result.selectTicketHandler(data);
        expect(dispatch).toHaveBeenCalledWith(actions.selectTicket(data));
      });
    });

    describe('cancelOrderHandler', () => {
      it('should be injected', () => {
        const dispatch = jest.fn();
        const result = mapDispatchToProps(dispatch);
        expect(result.cancelOrderHandler).toBeDefined();
      });

      it('should dispatch cancelOrder when called', () => {
        const dispatch = jest.fn();
        const result = mapDispatchToProps(dispatch);
        result.cancelOrderHandler();
        expect(dispatch).toHaveBeenCalledWith(actions.cancelOrder());
      });
    });

    describe('completeBookingHandler', () => {
      it('should be injected', () => {
        const dispatch = jest.fn();
        const result = mapDispatchToProps(dispatch);
        expect(result.completeBookingHandler).toBeDefined();
      });

      it('should dispatch completeBooking when called', () => {
        const dispatch = jest.fn();
        const result = mapDispatchToProps(dispatch);
        result.completeBookingHandler();
        expect(dispatch).toHaveBeenCalledWith(actions.completeBooking());
      });
    });
  });
});
