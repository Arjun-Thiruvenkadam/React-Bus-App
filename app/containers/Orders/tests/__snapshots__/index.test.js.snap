// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Orders /> should render and match the snapshot 1`] = `
<div
  class="Modal"
  style="visibility: hidden; opacity: 0;"
>
  <div
    class="Loader"
  >
    <h3>
      <span>
        Loading...
      </span>
    </h3>
  </div>
</div>
`;

exports[`<Orders /> should render and match the snapshot with spinner 1`] = `
<div
  class="Modal"
  style="visibility: hidden; opacity: 0;"
>
  <div
    class="Loader"
  >
    <h3>
      <span>
        Loading...
      </span>
    </h3>
  </div>
</div>
`;
